apps/dotcms-ui/src/app/api/services/dot-paginator-es-content/dot-paginator-es-content.service.ts

```diff
@@ -0,0 +1,176 @@
+        if (value !== null && value !== undefined) {
+            this.extraParams.set(name, value.toString());
+        }
+    }
+
+    /**
+     * Delete extra parameters of the eventual request.
+     * @param string name
+     *
+     * @memberof DotThemeSelectorComponent
+     */
+    public deleteExtraParams(name: string): void {
+        this.extraParams.delete(name);
+    }
+
+    /**
+     * Send a pagination request with url as base URL, if url is null or undefined then
+     * it use the url property value instead.
+     * Also it use the values of sortField, sortOrder in the Body request
+     * @param url base url
+     */
+    public get(url?: string): Observable<any> {
+        const queryParams = this.getESQuery(this.getObjectFromMap(this.extraParams));
+
+        return this.coreWebService
+            .requestView({
+                body: JSON.stringify(queryParams),
+                method: 'POST',
+                url: url || this.url
+            })
+            .pipe(
+                map((response: ResponseView<any>) => {
+                    this.totalRecords = response.entity?.resultsSize;
+                    return response.entity?.jsonObjectView?.contentlets || null;
+                }),
+                take(1)
+            );
+    }
+
+    /**
+     * request the  pageParam page
+     * @param number [pageParam=1] Page to request
+     * @returns Observable<any[]>
+     * @memberof PaginatorServic
+     */
+    public getPage(pageParam = 1): Observable<any[]> {
+        this.currentPage = pageParam;
+        return this.get();
+    }
+
+    /**
+     * Request the current page
+     * @returns Observable<any[]>
+     * @memberof PaginatorService
+     */
+    public getCurrentPage(): Observable<any[]> {
+        return this.getPage(this.currentPage);
+    }
+
+    /**
+     * Use the offset to request a page.
+     * @param number offset Offset to be request
+     * @returns Observable<any[]>
+     * @memberof PaginatorService
+     */
+    public getWithOffset(offset: number): Observable<any[]> {
+        const page = this.getPageFromOffset(offset);
+        return this.getPage(page);
+    }
+
+    private getESQuery(params: { [key: string]: any }): any {
+        const query = {
+            query: JSON.stringify(params).replace(/"|{|}|,/g, ' '),
+            sort: `${this._sortField || ''} ${this._sortOrder || ''}`,
+            limit: this.paginationPerPage,
+            offset: (this.currentPage - 1) * this.paginationPerPage
+        };
+
+        return query;
+    }
+
+    private getPageFromOffset(offset: number): number {
+        return parseInt(String(offset / this.paginationPerPage), 10) + 1;
+    }
+
+    private getObjectFromMap(map: Map<string, any>): { [key: string]: any } {
+        let result = Array.from(map).reduce(
+            (obj, [key, value]) => Object.assign(obj, { [key]: value }),
+            {}
+        );
+
+        return result;
+    }
+}
```

Comment:
I don't think this service should be a copy of the original paginator service because the ES API is different, if we need pagination specific properties for the table we add them in another service that could be the component store.